Framework/PythonInterface/plugins/algorithms/SCDAdjustSampleNorm.py

```diff
@@ -0,0 +1,246 @@
+        vanws = self.getProperty("VanadiumWorkspace").value
+        height = self.getProperty("DetectorHeightOffset").value
+        distance = self.getProperty("DetectorDistanceOffset").value
+        method = self.getProperty("OutputAsMDEventWorkspace").value
+
+        if method:
+            minvals = self.getProperty("MinValues").value
+            maxvals = self.getProperty("MaxValues").value
+            merge = self.getProperty("MergeInputs").value
+            wslist = []
+        else:
+            bin0 = self.getProperty("BinningDim0").value
+            bin1 = self.getProperty("BinningDim1").value
+            bin2 = self.getProperty("BinningDim2").value
+
+        out_ws = self.getPropertyValue("OutputWorkspace")
+        out_ws_name = out_ws
+
+        if load_van:
+            vanws = Load(vanadiumfile, StoreInADS=False)
+
+        has_multiple = True if len(datafiles) > 1 else False
+
+        # Default wavelength in WANDSCDtoQ if not set in the input file
+        wavelength = 1.488
+
+        for file in datafiles:
+            if load_files:
+                scan = Load(file, LoadHistory=False, StoreInADS=False)
+            else:
+                scan = mtd[file]
+
+            self.log().information("Processing file '{}'".format(file))
+
+            # If processing multiple files, append the base name to the given output name
+            if has_multiple:
+                if load_files:
+                    out_ws_name = out_ws + "_" + os.path.basename(file).strip(',.nxs')
+                else:
+                    out_ws_name = out_ws + "_" + file
+
+            exp_info = scan.getExperimentInfo(0)
+
+            # Adjust detector height and distance with new offsets
+            if height != 0.0 or distance != 0.0:
+                # Move all the instrument banks (MoveInstrumentComponents does not work on MDHisto Workspaces)
+                component = exp_info.componentInfo()
+                for bank in range(1, 4):
+                    index = component.indexOfAny("bank{}".format(bank))
+
+                    offset = V3D(0.0, height, distance)
+                    pos = component.position(index)
+
+                    offset += pos
+
+                    component.setPosition(index, offset)
+
+            # Get the wavelength from the file sample logs if it exists
+            if exp_info.run().hasProperty("wavelength"):
+                wavelength = exp_info.run().getProperty("wavelength").value
+
+            # Use ConvertHFIRSCDtoQ (and normalize van), or use ConvertWANDSCtoQ which handles normalization itself
+            if method:
+                van_norm = ReplicateMD(ShapeWorkspace=scan, DataWorkspace=vanws, StoreInADS=False)
+                van_norm = DivideMD(LHSWorkspace=scan, RHSWorkspace=van_norm, StoreInADS=False)
+                ConvertHFIRSCDtoMDE(InputWorkspace=scan, Wavelength=wavelength, MinValues=minvals,
+                                    MaxValues=maxvals, OutputWorkspace=out_ws_name)
```

Comment:
It looks like you are using the same OutputWorkspace for every file, does that work? Does it not override the workspace every time this runs before you merge them? I haven't tested this yet but it doesn't look correct.